payments-ui-core/src/test/java/com/stripe/android/ui/core/elements/CardNumberControllerTest.kt

```diff
@@ -466,21 +468,27 @@ internal class CardNumberControllerTest {
             }
         }
 
-        // Start by typing in an Amex card then a Mastercard
-        cardNumberController.onValueChange("37")
-        cardNumberController.onValueChange("372")
-        cardNumberController.onValueChange("")
-        cardNumberController.onValueChange("5555")
-
-        // Verify that AmericanExpress was reported once
-        val americanExpressReports = fakeDisallowedEventReporter.reportedBrands
-            .filter { it == CardBrand.AmericanExpress }
-        assertEquals(1, americanExpressReports.size, "AmericanExpress should be reported once")
-
-        // Verify that MasterCard was reported once
-        val masterCardReports = fakeDisallowedEventReporter.reportedBrands
-            .filter { it == CardBrand.MasterCard }
-        assertEquals(1, masterCardReports.size, "MasterCard should be reported once")
+        fakeDisallowedEventReporter.reportedBrands.test {
+            // Simulate entering "37" for American Express
+            cardNumberController.onValueChange("37")
+            // Simulate entering "372" (still American Express)
+            cardNumberController.onValueChange("372")
+            // Simulate clearing the input
+            cardNumberController.onValueChange("")
+            // Simulate entering "5555" for MasterCard
+            cardNumberController.onValueChange("5555")
```

Comment:
Can you add an `ensureNoEvents` or whatever the API is before this to ensure none of the previous events triggered the emission.

Reply:
Added expectNoEvents, I think that's it.